cmd/server/http/http_test.go

```diff
@@ -72,3 +73,42 @@ func TestAuthenticate(t *testing.T) {
 		})
 	}
 }
+
+func TestCommitMessageExtraction(t *testing.T) {
+	tt := []struct {
+		name          string
+		commitMessage string
+		commitInfo    commitInfo
+		err           error
+	}{
+		{
+			name:          "exact values",
+			commitMessage: "[test-service] artifact master-1234ds13g3-12s46g356g by Foo Bar\nSigned-off-by: Foo Bar <test@lunar.app>",
+			commitInfo: commitInfo{
+				AuthorEmail: "test@lunar.app",
+				AuthorName:  "Foo Bar",
+				Service:     "test-service",
+			},
+			err: nil,
+		},
+		{
+			name:          "missing signoff",
+			commitMessage: "[product] build something",
+			commitInfo:    commitInfo{},
+			err:           errors.New("not enough matches"),
+		},
+	}
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			info, err := extractAuthorFromCommit(tc.commitMessage)
+			if tc.err != nil {
+				assert.EqualError(t, errors.Cause(err), tc.err.Error(), "output error not as expected")
+			} else {
+				assert.NoError(t, err, "no output error expected")
+			}
+			assert.Equal(t, tc.commitInfo.AuthorName, info.AuthorName, "AuthorName not as expected")
+			assert.Equal(t, tc.commitInfo.Service, info.Service, "Service not as expected")
+			assert.Equal(t, tc.commitInfo.AuthorEmail, info.AuthorEmail, "AuthorEmail not as expected")
```

Comment:
If we in the future add additional fields to `commitInfo` they will not automatically be asserted for with this approach.
It could be reduced to just comparing the `commitInfo` structs directly and any new fields then gets included automatically.

```suggestion
			assert.Equal(t, tc.commitInfo, info, "commit info not as expected")
```

Reply:
You didn't push this?